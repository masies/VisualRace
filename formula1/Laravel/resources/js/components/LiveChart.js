import React from "react";
import Chart from "react-apexcharts";
import ApexCharts from "apexcharts";


class LiveChart extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      options: {
        chart: {
          fontFamily: 'Poppins, Helvetica, Arial, sans-serif',
          // background: '#353941',
          id: this.props.id,
          animations: {
            enabled: true,
            easing: 'linear',
            //   animateGradually: {
            //               enabled: true,
            //               delay: 100
            //           },
            dynamicAnimation: {
              enabled: true,
              speed: 200,
            },
          },

          toolbar: {
            show: false
          },
          zoom: {
            enabled: false
          }
        },
        colors: this.props.colors,
        stroke: {
          curve: "smooth",
          lineCap: 'round',
          width: 2,
        },
        title: {
          text: `${this.props.id} over Time [s]`,
          align: "center"
        },
        xaxis: {
          tickAmount: 5,
          tickPlacement: 'between',
          labels: {
            formatter: function (value) {
              return (value >= 0) ? Math.ceil(value) : "";
            }
          },
          type: 'numeric',
          range: 10,
          min: 5,
          title: {
            text: "Time [s]"
          },
        },
        yaxis: {
          tickAmount: 2,
          min: this.props.minYScale > 0 ? this.props.minYScale : 0,
          max: this.props.maxYScale,
          labels: {
            formatter: function (value) {
              return Math.ceil(value);
            }
          },
          title: {
            text: this.props.id
          },
        },
        legend: {
          show: true
        },
        tooltip: {
          enabled: false
        },
        theme: {
          mode: 'light',
          palette: 'palette5',
          monochrome: {
            enabled: false,
            color: '#255aee',
            shadeTo: 'light',
            shadeIntensity: 0.65
          },
        }
      },
      series: this.props.firstSeries
    };
  }

  componentDidMount() {
    this.state.series.forEach(driver => {
      if (!driver.chekced) {
        ApexCharts.exec(this.props.id, "hideSeries", driver.name)
      }
    })
  }

  shouldComponentUpdate(nextProps) {
    if (this.props.drivers.toString() !== nextProps.drivers.toString()) {
      nextProps.drivers.forEach(seriesName => {
        if (seriesName[0]) {
          ApexCharts.exec(this.props.id, "showSeries", seriesName[1])
        } else {
          ApexCharts.exec(this.props.id, "hideSeries", seriesName[1])
        }
      })
    }

    if (this.props.data.toString() !== nextProps.data.toString()) {
      let newSeries = []
      nextProps.data.forEach(data => {
        newSeries.push({
          name: data[0],
          data: [data[1]]
        })
      })
      ApexCharts.exec(this.props.id, "appendData", newSeries);
    }

    return false;
  }

  render() {
    return (
      <div id="chart">
        <Chart
          options={this.state.options}
          series={this.state.series}
          type="line"
          height="250"
          width="100%"
        />
      </div>
    );
  }
}

export default LiveChart;