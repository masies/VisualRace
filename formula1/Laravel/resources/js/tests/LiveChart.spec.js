import React from 'react';
import LiveChart from '../components/LiveChart';
import renderer from 'react-test-renderer';


describe('LiveChart', () => {
  it('should render the LiveChart Component correctly', () => {
    const component = renderer.create(<LiveChart max={0} firstSeries={[]} colors={[]} drivers={[]} maxYScale={160} minYScale={0} id={"Speed"} data={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})