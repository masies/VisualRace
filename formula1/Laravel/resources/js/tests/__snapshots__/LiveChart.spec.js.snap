// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveChart should render the LiveChart Component correctly 1`] = `
<div
  id="chart"
>
  <div
    height="250"
    options={
      Object {
        "chart": Object {
          "animations": Object {
            "dynamicAnimation": Object {
              "enabled": true,
              "speed": 200,
            },
            "easing": "linear",
            "enabled": true,
          },
          "fontFamily": "Poppins, Helvetica, Arial, sans-serif",
          "id": "Speed",
          "toolbar": Object {
            "show": false,
          },
          "zoom": Object {
            "enabled": false,
          },
        },
        "colors": Array [],
        "legend": Object {
          "show": true,
        },
        "stroke": Object {
          "curve": "smooth",
          "lineCap": "round",
          "width": 2,
        },
        "theme": Object {
          "mode": "light",
          "monochrome": Object {
            "color": "#255aee",
            "enabled": false,
            "shadeIntensity": 0.65,
            "shadeTo": "light",
          },
          "palette": "palette5",
        },
        "title": Object {
          "align": "center",
          "text": "Speed over Time",
        },
        "tooltip": Object {
          "enabled": false,
        },
        "xaxis": Object {
          "convertedCatToNumeric": false,
          "labels": Object {
            "formatter": [Function],
          },
          "min": 5,
          "range": 10,
          "tickAmount": 5,
          "tickPlacement": "between",
          "title": Object {
            "text": "Time",
          },
          "type": "numeric",
        },
        "yaxis": Object {
          "labels": Object {
            "formatter": [Function],
          },
          "max": 160,
          "min": 0,
          "tickAmount": 2,
          "title": Object {
            "text": "Speed",
          },
        },
      }
    }
    series={Array []}
    type="line"
    width="100%"
  />
</div>
`;
